compiler-rt/lib/scudo/standalone/secondary.h

```diff
@@ -122,18 +122,35 @@ bool mapSecondary(const Options &Options, uptr CommitBase, uptr CommitSize,
   Flags |= MAP_RESIZABLE;
   Flags |= MAP_ALLOWNOMEM;
 
-  const uptr MaxUnusedCacheBytes = MaxUnusedCachePages * getPageSizeCached();
+  const uptr PageSize = getPageSizeCached();
+  const uptr MaxUnusedCacheBytes = MaxUnusedCachePages * PageSize;
   if (useMemoryTagging<Config>(Options) && CommitSize > MaxUnusedCacheBytes) {
-    const uptr UntaggedPos = Max(AllocPos, CommitBase + MaxUnusedCacheBytes);
-    return MemMap.remap(CommitBase, UntaggedPos - CommitBase, "scudo:secondary",
-                        MAP_MEMTAG | Flags) &&
-           MemMap.remap(UntaggedPos, CommitBase + CommitSize - UntaggedPos,
-                        "scudo:secondary", Flags);
-  } else {
-    const uptr RemapFlags =
-        (useMemoryTagging<Config>(Options) ? MAP_MEMTAG : 0) | Flags;
-    return MemMap.remap(CommitBase, CommitSize, "scudo:secondary", RemapFlags);
+    if (SCUDO_TRUSTY) {
+      /*
```

Comment:
It's better to separate the logic. I.e.

```
if (SCUDO_TRUSTY) {
  // all the logic for trusty  
}

// all the others
if (useMemoryTagging<Config>(Options) && CommitSize > MaxUnusedCacheBytes) {
  // ...
} else {
  // ...
}
```

Reply:
I think that might require duplicating lines 151-153 for Trusty, would that be alright?